specs/configure_monitoring.adoc

```diff
@@ -0,0 +1,304 @@
+// vim: tw=79
+
+= Configuring collectd
+
+== Introduction
+
+Tendrl's monitoring stack uses the following components:
+
+* collectd
+  ** A daemon process that resides on tendrl managed nodes, collects
+     performance statistics and pushes them to time-series db for maintaining
+     historic data.
+  ** Everything in collectd is done in plugins.
+  ** Collectd activates those plugins that are loaded and/or configured in
+     collectd’s conf files.
+  ** Changes in conf files will be effected in collectd only on restart of
+     the collectd daemon.
+
+* graphite
+  ** Used by tendrl monitoring stack as time series db
+  ** It consists of three software components:
+    *** carbon - a high-performance service that listens for time-series data
+    *** whisper - a simple database library for storing time-series data
+    *** graphite-web - Graphite's user interface & API for rendering graphs
+        and dashboards
+  ** Metrics get fed into the stack via the Carbon service, which writes data
+     out to Whisper databases for long-term storage.
+
+== Problem Description
+
+Tendrl needs to configure collectd for it to start collecting the performance
+statistics at regular intervals of time. This process of configuring collectd
+should require no/minimal user intervention.
+
+== Use Cases
+
+The use cases addressed by this spec are:
+
+* Tendrl core and monitoring stack are installed and user starts node-agent on
+  nodes so that they can be managed by tendrl and the new node needs to be
+  monitored.
+* Tendrl core stack is installed and one/more nodes are managed by tendrl core.
+  Now the user decides to monitor his tendrl managed nodes by installing the
+  tendrl/performance_monitoring.
+* User plugs out the monitoring after using for a while.
+
+== Proposed change
+
+* The tendrl/performance_monitoring on start, checks the existence of
+  '/monitoring/defaults' directory in etcd and if its not present, it uploads
```

Comment:
@r0h4n Why this needs to be part of a flow definition? This is something that is supposed to happen at the start of the application(not triggered by user explicitly). Also, could you please add sample definition in the specs addressing #32